pytorch_lightning/loggers/wandb.py

```diff
@@ -46,12 +46,180 @@ class WandbLogger(LightningLoggerBase):
     r"""
     Log using `Weights and Biases <https://docs.wandb.ai/integrations/lightning>`_.
 
-    Install it with pip:
+    **Installation and set-up**
+
+    Install with pip:
 
     .. code-block:: bash
 
         pip install wandb
 
+    Create a `WandbLogger` instance:
+
+    .. code-block:: python
+
+        from pytorch_lightning.loggers import WandbLogger
+
+        wandb_logger = WandbLogger(project="MNIST")
+
+    Pass the logger instance to the `Trainer`:
+
+    .. code-block:: python
+
+        trainer = Trainer(logger=wandb_logger)
+
+    A new W&B run will be created when training starts if you have not created one manually before with `wandb.init()`.
+
+    **Log metrics**
+
+    Log from :class:`~pytorch_lightning.core.lightning.LightningModule`:
+
+    .. code-block:: python
+
+        class LitModule(LightningModule):
+            def training_step(self, batch, batch_idx):
+                self.log("train/loss", loss)
+
+    Use directly wandb module:
+
```

Comment:
I won't recommend to present the wandb client directly. This would be confusing the users. They can get this information from wandb doc directly.

Reply:
Sure I could remove it but the main idea here was:
* for custom objects not supported yet (I will add more in a follow-up PR).
* logging metrics before the start of the run